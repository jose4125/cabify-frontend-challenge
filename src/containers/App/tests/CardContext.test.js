import React from 'react';
import { shallow } from 'enzyme';

import CardProvider, { CardContext } from '../CardContext';

describe('CardContext', () => {
  let state;
  const wrapper = shallow(
    <CardProvider
      value={{
        state: state
      }}
    >
      <div>test</div>
    </CardProvider>
  );

  beforeEach(() => {
    state = {
      fullName: '',
      jobDescription: '',
      prefix: '',
      phoneNumber: '',
      email: '',
      website: '',
      address: ''
    };
  });
  it('should render with out craching', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have at least one children', () => {
    expect(wrapper.children().length).toBe(1);
  });

  it('should the provider have a value attribute', () => {
    expect(wrapper.prop('value').state).toEqual(state);
  });

  it('should the provider update the full name in the state', () => {
    const fullName = 'jose david';
    const key = 'fullName';
    wrapper.prop('value').handleChange(fullName, key);

    expect(wrapper.state(key)).toBe(fullName);
  });

  it('should the provider update the prefix in the state', () => {
    const prefix = '+57';
    const key = 'prefix';
    wrapper.prop('value').handlePrefix(prefix, key);

    expect(wrapper.state(key)).toBe(prefix);
  });

  it('should the provider reset the state', () => {
    const prefix = '+57';
    const state = {
      fullName: '',
      jobDescription: '',
      prefix: prefix,
      phoneNumber: '',
      email: '',
      website: '',
      address: ''
    };
    wrapper.prop('value').handleReset(prefix);

    expect(wrapper.state()).toEqual(state);
  });
});
