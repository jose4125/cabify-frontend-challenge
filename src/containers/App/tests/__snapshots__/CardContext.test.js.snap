// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContext should render with out craching 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardProvider
    value={
      Object {
        "state": undefined,
      }
    }
  >
    <div>
      test
    </div>
  </CardProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div>
        test
      </div>,
      "value": Object {
        "handleChange": [Function],
        "handlePrefix": [Function],
        "handleReset": [Function],
        "state": Object {
          "address": "",
          "email": "",
          "fullName": "",
          "jobDescription": "",
          "phoneNumber": "",
          "prefix": "",
          "website": "",
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
      },
      "ref": null,
      "rendered": "test",
      "type": "div",
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_changedBits": 0,
        "_changedBits2": 0,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "_defaultValue": undefined,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div>
          test
        </div>,
        "value": Object {
          "handleChange": [Function],
          "handlePrefix": [Function],
          "handleReset": [Function],
          "state": Object {
            "address": "",
            "email": "",
            "fullName": "",
            "jobDescription": "",
            "phoneNumber": "",
            "prefix": "",
            "website": "",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_changedBits": 0,
          "_changedBits2": 0,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": undefined,
          "_currentValue2": undefined,
          "_defaultValue": undefined,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
